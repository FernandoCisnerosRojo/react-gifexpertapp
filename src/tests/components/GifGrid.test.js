import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid Tests', () => {
    test('should match snapshot while loading', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={'car'} />);

        expect(wrapper).toMatchSnapshot();
    })

    test('should have gifs loaded', () => {
        const gifs = [
            {
                id: 'abc',
                url: 'https://localhost/test.gif',
                title: 'test title'
            },
            {
                id: 'abcd',
                url: 'https://localhost/test.gif',
                title: 'test title'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={'car'} />);

        // expect(wrapper).toMatchSnapshot();u
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })


})
