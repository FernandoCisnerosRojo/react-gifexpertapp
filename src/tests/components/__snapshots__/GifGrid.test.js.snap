// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Tests should match snapshot while loading 1`] = `
<Fragment>
  <h3>
    car
  </h3>
  <p>
    'Cargando...'
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
